### ISSUE TITLE
Chairs (etc.) in charters become stale

### ISSUE THREAD

**Issue by samuelweiler** [w3c/charter-drafts#259]
Listing chairs in charters creates redundant info that must be manually maintained.  Can we remove that info from the charter (and instead refer to a DB-backed source of truth)?  

I'm thinking of something like the data shown in:
https://www.w3.org/Member/Groups

**Comment by samuelweiler** [comment]
Lots of discussion about whether this is a problem and how to solve it in https://github.com/w3c/charter-drafts/pull/261

Renaming this issue to capture the problem rather than the proposed solution.

**Comment by dontcallmedom** [comment]
I think operationally, the way to ensure that charters be updated in place is to ask the Comm Team to refrain from sending the announcement until the charter has been updated.

**Comment by wseltzer** [comment]
Closing with reference to the operational fixes, rather than changes to the charter template.

**Comment by samuelweiler** [comment]
When I checked four weeks ago, I found:

- 4 WGs or IGs where the chair(s) in the charter did not match the DB
- 3 team contact mismatches
- 2 charter date mismatches (plus one typo)

I pointed all out to the responsible team contacts, CC'ing w3t-comm on some of them.  Some were fixed.  Meanwhile, new errors appeared.  Now, four weeks later, we have:

- 5 chair mismatches (2 old, 3 new)
- 2 team contact mismatches (both old)
- 3 charter date mismatches (2 old, 1 new)

In summary: despite the operational fixes discussed above and an effort to fix the specific errors from one point in time, new errors have appeared, and we are still at about the same overall error rate as before.

I am not naming names.  I see this more as a failure of our systems than as a failure of individuals.  

@wseltzer Do you still want to leave this GH issue closed?

(n.b. at least one of the chair mismatches was explained to me as "cannot be fixed".)

**Comment by dontcallmedom** [comment]
@samuelweiler is there any evidence that the operational fix was put in place (i.e. that it was asked from the Comm Team and the comm Team couldn't or wouldn't?)

**Comment by samuelweiler** [comment]
@dontcallmedom I had thought so but, as I look back, perhaps not.  @koalie ?

**Comment by koalie** [comment]
> @dontcallmedom I had thought so but, as I look back, perhaps not. @koalie ?

Comm was cc'ed on some of the issues you had spotted, not all as far as you wrote.

**Comment by koalie** [comment]
Proposal: One operational change we could adopt is the generalization of charters edits by comm in the wake of comm sending the various related announcements. Thoughts?

**Comment by samuelweiler** [comment]
> Comm was cc'ed on some of the issues you had spotted, not all as far as you wrote.

Correct.  I think Dom was asking about the operational fixes - to reduce new errors.

**Comment by koalie** [comment]
> > Comm was cc'ed on some of the issues you had spotted, not all as far as you wrote.
> 
> Correct. I think Dom was asking about the operational fixes - to reduce new errors.

I'm sorry, I can't recall at the moment any proposed generic operational fixes.

**Comment by samuelweiler** [comment]
> Proposal: One operational change we could adopt is the generalization of charters edits by comm in the wake of comm sending the various related announcements. Thoughts?

Dom [proposed](https://github.com/w3c/charter-drafts/issues/259#issuecomment-641385632) doing the charter changes before the announcements:   "...ask the Comm Team to refrain from sending the announcement until the charter has been updated."

**Comment by koalie** [comment]
> > Proposal: One operational change we could adopt is the generalization of charters edits by comm in the wake of comm sending the various related announcements. Thoughts?
> 
> Dom [proposed](https://github.com/w3c/charter-drafts/issues/259#issuecomment-641385632) doing the charter changes before the announcements: "...ask the Comm Team to refrain from sending the announcement until the charter has been updated."

I was no aware of Dom's proposal. Thanks for calling my attention to it.

My most recent counter proposal probably scales better. 
Unless it's a nightmare for us between the various ways pages are maintained (cvs vs. gh), OR unless team contacts object to letting us do so.

**Comment by samuelweiler** [comment]
> I was no aware of Dom's proposal. Thanks for calling my attention to it.

You are welcome.  I guess that answers Dom's question about whether an operational fix was put in place.  :-). 

What support, if any, do you need to make the change?  (And, oh, wouldn't it be nice to have those pages maintained in a common way?)

**Comment by koalie** [comment]
> > I was no aware of Dom's proposal. Thanks for calling my attention to it.
> 
> You are welcome. I guess that answers Dom's question about whether an operational fix was put in place. :-).
> 
> What support, if any, do you need to make the change? (And, oh, wouldn't it be nice to have those pages maintained in a common way?)

First, we need the vetting from @plehegar and @wseltzer that we can put this in place. That includes polling the team contacts. If there's broad agreement we should indeed proceed this way, I'll work with @xueyuanjia to make it happen (she primarily and I are the ones to take the action.)

Edit: And then, we need to announce it and document it in our handbook.

**Comment by xueyuanjia** [comment]
+1 to charters accurate up-to-date. 

I (Comm: @koalie and me) can take to update charters timely when there are Comm announcements (re-charter, group extension, chair (re) appointment). 

However, it's hard for Comm to track changes that don't involve announcements, e.g., new Team Contacts, FTE changes or editorial changes (copyright, error links, typos). I see them better in Team Contacts' hands.

Chairs mismatches: occasionally staffs (un)mark Chairs in the DB without announcements first, which should be avoided per our Process. 

Team Contacts mismatches: people mark new Team Contacts in the DB without updating charters at the same time...maybe allow Team Contacts to flexibly update charters whenver this happens, instead of waiting until rechater/extension, can improve this? which means:

* Comm to update charters following announcements
* Team Contact to update charter as needed (when no announcement required)

Another issue, some charters are missing "History" section to record relevant changes. To be consistent and also make future work easier, it'll be helpful to have this section added. E.g.,
* [EOWG](https://www.w3.org/WAI/EO/charter2020#about): has History, only lists main addtions from previouse chaters but no extension/chairs changes recorded
* [JSON-LD WG](https://www.w3.org/2018/03/jsonld-wg-charter.html#about): no History
* [Media WG](https://www.w3.org/2019/05/media-wg-charter.html#about): no History
* [WebPlat WG](https://www.w3.org/2017/08/webplatform-charter.html#about): has Chages section, but seems need to be directed to [WebApps WG](https://www.w3.org/2019/05/webapps-charter.html#about) (?)
* [WAI IG](https://www.w3.org/WAI/IG/charter6.html#about): no History
* [Web Payment Security IG](https://www.w3.org/securepay/charter#about): no History

My two cents above.

**Comment by chaals** [comment]
Thanks for the detailed explanations @xueyuanjia 

These are all team operational issues, and the team should put in place processes to sort them out. The history sections are important, just as they are in specifications.

**Comment by samuelweiler** [comment]
Four weeks later, it's time for another report.  Here's the [last one](https://github.com/w3c/charter-drafts/issues/259#issuecomment-654909629).

- 4 chair mismatches (3 old, 1 new) + 1 chair affiliation mismatch (new? or missed before)
- 1 team contact mismatch (old) 
- 1 charter date mismatch (old)

This is out of 34 WGs and 10 IGs. 

As before, one of the chair mismatches was explained to me as "cannot be fixed".

**Comment by samuelweiler** [comment]
Four weeks later:

- 3 chair mismatches (2 old, 1 new; one of the old ones "can't be fixed")
- 2 team contact mismatches (1 old, 1 new)
- 0 charter date mismatches
- 1 group listed on the WG page that isn't a WG.  (Voice Interaction workshop PC.) [fixed same day reported]

(Out of 36 WGs and 10 IGs.)

Details:

- Web Applications Working Group chairs - new
- WPSIG chairs - old (can't be fixed)
- Web Platform WG chairs - old
- Web Performance WG team contact - new [fixed same day reported]
- i18n WG team contact - old

**Comment by chaals** [comment]
What do you mean "can't be fixed"? This is a document on the Web, right?

**Comment by samuelweiler** [comment]
> What do you mean "can't be fixed"? This is a document on the Web, right?

The participants page at https://www.w3.org/groups/ig/securepay/participants is produced from W3C's database; it is not a static doc.

As it was explained to me, the Web Payments Security IG "involves three organizations. The DBWG database only reflects those who have joined from the W3C. The tool does not reflect that the chairs are participating as chairs of their respective organizations."

In other words, this special case has not been special-cased in the tooling.

**Comment by samuelweiler** [comment]
Another 4 weeks, another report:

3 chair mismatches (2 old, 1 new; one of the old ones "can't be fixed")
2 team contact mismatches (1 old, 1 new)

So exactly the same total number of errors as 4 weeks ago.

**Comment by samuelweiler** [comment]
(Even TPAC isn't stopping the generation of this report!)

Good news!  No new discrepancies appeared in this every-four-week audit.  One item has been fixed.

3 chair mismatches (all old, one "can't be fixed")
1 team contact mismatch (old)

Bad news: with only 37 WGs and 10 IGs, we still have >5% with a discrepancy.

**Comment by samuelweiler** [comment]
The list of discrepancies is the same as it was on 27 October 2020.  All four have been on the list since at least September. Two of the four have been on the list since I started doing these audits in early June.
